src/Components/CommentAdder.jsx

```diff
@@ -0,0 +1,46 @@
+import { useContext, useState } from "react";
+import { UserContext } from "../contexts/UserContext";
+import { postComment } from "../utils/axios";
+
+const CommentAdder = ({ article_id, setComments }) => {
+  const { user } = useContext(UserContext);
+  const [newComment, setNewComment] = useState("");
+  const [isPosting, setIsPosting] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setIsPosting(true);
+    postComment(article_id, user.username, newComment)
+      .then((response) => {
+        setComments((currComments) => {
+          return [response, ...currComments];
+        });
+      })
+      .then(() => {
+        setNewComment("");
+        setIsPosting(false);
+      });
+  };
+
+  return (
+    <form className="comment-adder" onSubmit={handleSubmit}>
+      <label htmlFor="new-comment-input">
+        comment as <span className="comment-user">{user.username}</span>
+      </label>
+      <textarea
+        id="new-comment-input"
+        className="new-comment-input"
+        placeholder="add a new comment"
+        rows="10"
+        required
+        value={newComment}
+        onChange={(event) => setNewComment(event.target.value)}
+      ></textarea>
+      <button className="comment-adder-button" disabled={isPosting}>
+        comment
+      </button>
+    </form>
+  );
+};
```

Comment:
Nice work 🙂 it would be nice to get some kind of feedback for the user if the postedComment isn't successful. I get the error in the console but a message displayed would be helpful 👀